mlflow/tracking/client.py

```diff
@@ -706,6 +708,14 @@ def log_metric(
         :param timestamp: Time when this metric was calculated. Defaults to the current system time.
         :param step: Integer training step (iteration) at which was the metric calculated.
                      Defaults to 0.
+        :param synchronous: [Experimental] Defaults to True with no behavior change.
+                             Indicates if the metric would be logged in synchronous fashion or not.
+                            When it is True this call would be blocking call and offers immediate
+                             consistency of the metric upon returning.
+                            When this value is set to False, metric would be logged in async
+                             fashion. Backing provider gurantees that metrics are accepted
+                             into system but would persist them with some time delay -
+                             'near-real time'/'eventual' consistency.
```

Comment:
I think we should simplify this. We also shouldn't refer 'behavior change', since that won't be meaningful to users.

```suggestion
        :param synchronous: *Experimental* If True, blocks until the metric is logged successfully. If False,
                                             logs the metric asynchronously and returns a future representing the logging operation.
```